fix(vtt): multiline support

frost768 opened the pull request:
Instead of using `\s` using the literal character `[  ]*` matches with the styles if there is any and prevents matching the first line of multiple line captions

leranjun commented:
@frost768 Thanks a lot for opening the PR! Could you please provide an example test case that this PR is trying to fix?

frost768 commented:
> @frost768 Thanks a lot for opening the PR! Could you please provide an example test case that this PR is trying to fix?

I have explained the test case in issue #2.

leranjun commented on lib/format/vtt.ts:
```diff
@@ -38,7 +38,7 @@ const parse = (content: string, options: ParseOptions) => {
         // WebVTT data
         const regex =
-            /^([^\r\n]+\r?\n)?((?:\d{1,2}:)?\d{1,2}:\d{1,2}(?:[.,]\d{1,3})?)\s*-->\s*((?:\d{1,2}:)?\d{1,2}:\d{1,2}(?:[.,]\d{1,3})?)\s?.*\r?\n([\s\S]*)$/;
+            /^([^\r\n]+\r?\n)?((?:\d{1,2}:)?\d{1,2}:\d{1,2}(?:[.,]\d{1,3})?)\s*-->\s*((?:\d{1,2}:)?\d{1,2}:\d{1,2}(?:[.,]\d{1,3})?)[ ]*.*\r?\n([\s\S]*)$/;
```
Perhaps instead of specifying only space, we can use `[^\S\r\n]`? In that way it should match `\s` without new lines.

This seems to be working for me: `^([^\r\n]+\r?\n)?((?:\d{1,2}:)?\d{1,2}:\d{1,2}(?:[.,]\d{1,3})?)\s*-->\s*((?:\d{1,2}:)?\d{1,2}:\d{1,2}(?:[.,]\d{1,3})?)[^\S\r\n]?.*\r?\n([\s\S]*)$`

leranjun replied:
By the way, I'm thinking of refactoring all RegExps to use named capturing groups since it's easier to read, but maybe after merging this PR.

frost768 replied:
That's a better approach.

leranjun reviewed (changes requested):
Apologies for missing #2 completely! Here is my thought on this :)

leranjun commented:
Also, would be great if you can edit the commit message to use conventional commit, just to keep things tidy :)

frost768 commented:
> Also, would be great if you can edit the commit message to use conventional commits, just to keep things tidy :)

Of course, I will do it.

There is also an another thing that I've changed but didn't include in this pr. I see that you are splitting captions by new line and join them into a single line string with a space. You may have a reason to do this but in the example case i gave it can cause some issues for readers. Changing
```
- Hi
- Hey
```
with `- Hi - Hey` doesn't seem right. Because of this, I also removed the replacement of new lines in captions in an another branch.

leranjun reviewed (approved):
LGTM - thank you very much for your contribution!

For the newline joining issue you mentioned, I think that segment is from the old subsrt package. Could you please open a separate issue for that? If you have time to work on a PR, that would be wonderful. If not, I'll have a look when I have time :)
